pkg/cloud/aws/actuators/machine/utils.go

```diff
@@ -141,30 +141,40 @@ func terminateInstances(client awsclient.Client, instances []*ec2.Instance) erro
 func ProviderConfigFromMachine(client client.Client, machine *clusterv1.Machine, codec *providerconfigv1.AWSProviderConfigCodec) (*providerconfigv1.AWSMachineProviderConfig, error) {
 	var providerConfig runtime.RawExtension
 
-	if machine.Spec.ProviderConfig.Value == nil && machine.Spec.ProviderConfig.ValueFrom == nil {
+	// TODO(jchaloup): Remove providerConfig once all consumers migrate to providerSpec
+	providerSpec := machine.Spec.ProviderConfig
+	// providerSpec has higher priority over providerConfig
+	if machine.Spec.ProviderSpec.Value != nil || machine.Spec.ProviderSpec.ValueFrom != nil {
```

Comment:
The expression changed from `==`, to `!=` -- assuming this was intentional.

Reply:
The expressions is still there, just represented as `if providerSpec.Value == nil && providerSpec.ValueFrom == nil {`. This expression makes sure ProviderSpec is used in case at least `.Value` or `.ValueFrom` is set. Otherwise, it falls down to `.providerConfig` field. So yes, it is intentional.